reporter/src/funTest/kotlin/reporters/SpdxDocumentReporterTest.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.reporter.reporters
+
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.shouldBe
+
+import java.io.File
+
+import org.ossreviewtoolkit.model.Environment
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.reporter.reporters.SpdxDocumentReporter.FileFormat
+import org.ossreviewtoolkit.spdx.SpdxModelSerializer.fromJson
+import org.ossreviewtoolkit.spdx.SpdxModelSerializer.fromYaml
+import org.ossreviewtoolkit.spdx.model.SpdxDocument
+import org.ossreviewtoolkit.utils.ORT_NAME
+import org.ossreviewtoolkit.utils.test.patchExpectedResult
+import org.ossreviewtoolkit.utils.test.readOrtResult
+
+class SpdxDocumentReporterTest : WordSpec({
+    "SpdxDocumentReporter" should {
+        "create the expected JSON SPDX document" {
+            val ortResult = readOrtResult("src/funTest/assets/static-html-reporter-test-input.yml")
+
+            val jsonSpdxDocument = generateReport(ortResult, FileFormat.JSON)
+
+            jsonSpdxDocument shouldBe patchExpectedResult(
+                "src/funTest/assets/spdx-document-reporter-expected-output.spdx.yml",
```

Comment:
This should be the `.json` file, or?

Reply:
I wonder how the JSON test could have failed when adding the final newline to the expected result if you were never using that file...

Reply:
> I wonder how the JSON test could have failed when adding the final newline to the expected result if you were never using that file...

That's a different story.